src/main/java/org/jenkinsci/plugins/rundeck/OptionProvider.java

```diff
@@ -161,20 +176,35 @@ public void doBuild(StaplerRequest request, StaplerResponse response) throws IOE
         // add optional references to last / lastStable / lastSuccessful builds
         if (Boolean.valueOf(request.getParameter("includeLastStableBuild"))) {
             Run<?, ?> build = project.getLastStableBuild();
+            try {
+                build.checkPermission(ARTIFACTS);
+            }catch (Exception e){
+                response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
+            }
```

Comment:
Since you're always doing this check before calling `findArtifact()` maybe you should simply move it inside the method and return nothing if it's false

Reply:
sound good